File: C++/firstUniqChar.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        std::unordered_map<char, int> charsCount;
        for (int i = 0; i < s.size(); ++i) {
            ++charsCount[s[i]];
        }
        for(int i = 0; i < s.size(); ++i) {
            if (charsCount.find(s[i])->second == 1) {
                return i;
            }
        }
        return -1;
    }
};

File: Typescript/firstUniqChar.ts
function firstUniqChar(s: string): number {
    const charsCount = new Map<string, number>();

    for (let i = 0; i < s.length; i++) {
        const char = s[i];
        if (charsCount.has(char)) {
            charsCount.set(char, charsCount.get(char)! + 1);
        } else {
            charsCount.set(char, 1);
        }
    }

    for (let i = 0; i < s.length; i++) {
        if (charsCount.get(s[i]) === 1) {
            return i;
        }
    }

    return -1;
}
